go-carbon with carbonserver module enabled can't run on raspberry pi

compiled with 
go version go1.9 linux/amd64
by running
GOOS=linux GOARCH=arm GOARM=5 go build -v github.com/lomik/go-carbon
run as /home/pi/carbon/go-carbon -daemon -config /home/pi/carbon/go-carbon.conf 

while carbonserver enabled
```
[common]
user = "pi"
graph-prefix = "carbon.agents.{host}"
metric-endpoint = "local"
metric-interval = "1m0s"
max-cpu = 2

[whisper]
data-dir = "/mnt/carbon/whisper"
schemas-file = "/home/pi/carbon/storage-schemas.conf"
workers = 4
max-updates-per-second = 10
sparse-create = false
enabled = true

[cache]
max-size = 100000
write-strategy = "noop"

[udp]
listen = ":2003"
enabled = false
log-incomplete = false
buffer-size = 0

[tcp]
listen = ":2003"
enabled = true
buffer-size = 0

[pickle]
listen = ":2004"
max-message-size = 67108864
enabled = false
buffer-size = 0

[carbonlink]
listen = "127.0.0.1:7002"
enabled = false
read-timeout = "30s"

[grpc]
listen = "127.0.0.1:7003"
enabled = false

[carbonserver]
listen = "127.0.0.1:8080"
enabled = true
buckets = 10
metrics-as-counters = false
read-timeout = "60s"
write-timeout = "60s"
query-cache-enabled = false
query-cache-size-mb = 10
find-cache-enabled = false
trigram-index = false
scan-frequency = "5m0s"
max-globs = 100
fail-on-max-globs = false
graphite-web-10-strict-mode = true
internal-stats-dir = ""
stats-percentiles = [99, 98, 95, 75, 50]

[dump]
enabled = false
path = "/var/lib/graphite/dump/"
restore-per-second = 0

[pprof]
listen = "localhost:7007"
enabled = false

[[logging]]
logger = ""
# Log output: filename, "stderr", "stdout", "none", "" (same as "stderr")
file = "stdout"
level = "debug"
encoding = "mixed"
encoding-time = "iso8601"
encoding-duration = "seconds"


``` 
```
pi@pi:~/carbon $ /home/pi/carbon/go-carbon -daemon -config /home/pi/carbon/go-carbon.conf 
[2017-10-13T22:44:36.326+0200] INFO [carbonserver] starting carbonserver {"whisperData": "/mnt/carbon/whisper", "maxGlobs": 100, "scanFrequency": "5m0s"}
[2017-10-13T22:44:36.326+0200] INFO [carbonserver] listening on 127.0.0.1:8080 {}
[2017-10-13T22:44:36.326+0200] INFO [main] started {}
[2017-10-13T22:44:36.425+0200] DEBUG [cache] WriteoutQueue.nextRebuildOnce.Do {"now": "2017-10-13 22:44:36.425522202 +0200 CEST m=+0.213451166", "next": "0001-01-01 00:00:00 +0000 UTC"}
[2017-10-13T22:44:36.426+0200] DEBUG [cache] WriteoutQueue.nextRebuildOnce.Do {"now": "2017-10-13 22:44:36.426640632 +0200 CEST m=+0.214569492", "next": "2017-10-13 22:44:36.526623757 +0200 CEST m=+0.314552565"}
[2017-10-13T22:44:36.426+0200] DEBUG [cache] WriteoutQueue sleep before rebuild {"sleepTime": "99.983073ms"}
----skip
[2017-10-13T22:45:36.099+0200] DEBUG [cache] WriteoutQueue.nextRebuildOnce.Do {"now": "2017-10-13 22:45:36.099538479 +0200 CEST m=+59.887467860", "next": "2017-10-13 22:45:36.199496344 +0200 CEST m=+59.987425569"}
[2017-10-13T22:45:36.099+0200] DEBUG [cache] WriteoutQueue sleep before rebuild {"sleepTime": "99.957709ms"}
[2017-10-13T22:45:36.201+0200] DEBUG [cache] WriteoutQueue.nextRebuildOnce.Do {"now": "2017-10-13 22:45:36.201190219 +0200 CEST m=+59.989119652", "next": "2017-10-13 22:45:36.301146365 +0200 CEST m=+60.089075590"}
[2017-10-13T22:45:36.201+0200] DEBUG [cache] WriteoutQueue sleep before rebuild {"sleepTime": "99.955938ms"}
[2017-10-13T22:45:36.303+0200] DEBUG [cache] WriteoutQueue.nextRebuildOnce.Do {"now": "2017-10-13 22:45:36.303549037 +0200 CEST m=+60.091478470", "next": "2017-10-13 22:45:36.403495913 +0200 CEST m=+60.191425137"}
[2017-10-13T22:45:36.303+0200] DEBUG [cache] WriteoutQueue sleep before rebuild {"sleepTime": "99.946667ms"}
[2017-10-13T22:45:36.327+0200] INFO [stat] collect {"endpoint": "local", "metric": "carbon.agents.pi.runtime.GOMAXPROCS", "value": 2}
[2017-10-13T22:45:36.327+0200] INFO [stat] collect {"endpoint": "local", "metric": "carbon.agents.pi.runtime.NumGoroutine", "value": 16}
[2017-10-13T22:45:36.327+0200] INFO [stat] collect {"endpoint": "local", "metric": "carbon.agents.pi.cache.size", "value": 1}
[2017-10-13T22:45:36.328+0200] INFO [stat] collect {"endpoint": "local", "metric": "carbon.agents.pi.cache.metrics", "value": 3}
[2017-10-13T22:45:36.328+0200] INFO [stat] collect {"endpoint": "local", "metric": "carbon.agents.pi.cache.maxSize", "value": 100000}
[2017-10-13T22:45:36.329+0200] INFO [stat] collect {"endpoint": "local", "metric": "carbon.agents.pi.cache.queries", "value": 0}
[2017-10-13T22:45:36.329+0200] INFO [stat] collect {"endpoint": "local", "metric": "carbon.agents.pi.cache.overflow", "value": 0}
[2017-10-13T22:45:36.329+0200] INFO [stat] collect {"endpoint": "local", "metric": "carbon.agents.pi.cache.queueBuildCount", "value": 589}
[2017-10-13T22:45:36.330+0200] INFO [stat] collect {"endpoint": "local", "metric": "carbon.agents.pi.cache.queueBuildTimeMs", "value": 567}
[2017-10-13T22:45:36.330+0200] INFO [stat] collect {"endpoint": "local", "metric": "carbon.agents.pi.cache.queueWriteoutTime", "value": 0}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x4 pc=0x11540]

goroutine 33 [running]:
sync/atomic.loadUint64(0x10fd82bc, 0x37ca8, 0x420adc)
	/opt/go/src/sync/atomic/64bit_arm.go:10 +0x3c
github.com/lomik/go-carbon/helper.SendAndSubstractUint64(0x5a3c0f, 0xf, 0x10fd82bc, 0x1116e3e0)
	/home/azhiltsov/go/src/github.com/lomik/go-carbon/helper/atomic_utils.go:8 +0x1c
github.com/lomik/go-carbon/carbonserver.(*CarbonserverListener).Stat(0x10fd8200, 0x1116e3e0)
	/home/azhiltsov/go/src/github.com/lomik/go-carbon/carbonserver/carbonserver.go:1574 +0x150
github.com/lomik/go-carbon/carbon.NewCollector.func4.1()
	/home/azhiltsov/go/src/github.com/lomik/go-carbon/carbon/collector.go:155 +0x5c
github.com/lomik/go-carbon/carbon.(*Collector).collect(0x10fd2e40)
	/home/azhiltsov/go/src/github.com/lomik/go-carbon/carbon/collector.go:205 +0x3c
github.com/lomik/go-carbon/carbon.NewCollector.func6(0x10f16640)
	/home/azhiltsov/go/src/github.com/lomik/go-carbon/carbon/collector.go:195 +0x60
github.com/lomik/go-carbon/helper.(*Stoppable).StartFunc.func1.1(0x10f0ce80, 0x10f16640, 0x10fd2e40)
	/home/azhiltsov/go/src/github.com/lomik/go-carbon/helper/stoppable.go:42 +0x24
created by github.com/lomik/go-carbon/helper.(*Stoppable).StartFunc.func1
	/home/azhiltsov/go/src/github.com/lomik/go-carbon/helper/stoppable.go:41 +0x64
```

--- a/carbonserver/carbonserver.go
+++ b/carbonserver/carbonserver.go
@@ -88,12 +88,11 @@ type metricStruct struct {
 	QueryCacheMiss       uint64
 	FindCacheHit         uint64
 	FindCacheMiss        uint64
+}
 
-	requestsTimes struct {
-		sync.RWMutex
-
-		list []int64
-	}
+type requestsTimes struct {
+	sync.RWMutex
+	list []int64
 }
 
 const (
@@ -253,9 +252,10 @@ type CarbonserverListener struct {
 	fileIdx      atomic.Value
 	fileIdxMutex sync.Mutex
 
-	metrics     metricStruct
-	exitChan    chan struct{}
-	timeBuckets []uint64
+	metrics       *metricStruct
+	requestsTimes requestsTimes
+	exitChan      chan struct{}
+	timeBuckets   []uint64
 
 	db *leveldb.DB
 }
@@ -284,6 +284,7 @@ type fileIndex struct {
 func NewCarbonserverListener(cacheGetFunc func(key string) []points.Point) *CarbonserverListener {
 	return &CarbonserverListener{
 		// Config variables
+		metrics:           &metricStruct{},
 		metricsAsCounters: false,
 		cacheGet:          cacheGetFunc,
 		logger:            zapwriter.Logger("carbonserver"),
@@ -1619,10 +1620,10 @@ func (listener *CarbonserverListener) Stat(send helper.StatCallback) {
 
 	// Computing response percentiles
 	if len(listener.percentiles) > 0 {
-		listener.metrics.requestsTimes.Lock()
-		list := listener.metrics.requestsTimes.list
-		listener.metrics.requestsTimes.list = make([]int64, 0, len(list))
-		listener.metrics.requestsTimes.Unlock()
+		listener.requestsTimes.Lock()
+		list := listener.requestsTimes.list
+		listener.requestsTimes.list = make([]int64, 0, len(list))
+		listener.requestsTimes.Unlock()
 		if len(list) == 0 {
 			for _, p := range listener.percentiles {
 				send(fmt.Sprintf("request_time_%vth_percentile_ns", p), 0)
@@ -1835,9 +1836,9 @@ func (listener *CarbonserverListener) bucketRequestTimes(req *http.Request, t ti
 	ms := t.Nanoseconds() / int64(time.Millisecond)
 
 	if len(listener.percentiles) > 0 {
-		listener.metrics.requestsTimes.Lock()
-		listener.metrics.requestsTimes.list = append(listener.metrics.requestsTimes.list, t.Nanoseconds())
-		listener.metrics.requestsTimes.Unlock()
+		listener.requestsTimes.Lock()
+		listener.requestsTimes.list = append(listener.requestsTimes.list, t.Nanoseconds())
+		listener.requestsTimes.Unlock()
 	}
 
 	bucket := int(math.Log(float64(ms)) * math.Log10E)

--- a/carbonserver/carbonserver_test.go
+++ b/carbonserver/carbonserver_test.go
@@ -259,6 +259,7 @@ func testFetchSingleMetricCommon(t *testing.T, test *FetchTest) {
 		whisperData: path,
 		cacheGet:    cache.Get,
 		logger:      zap.NewNop(),
+		metrics:     &metricStruct{},
 	}
 	precision := 0.000001
 
@@ -355,6 +356,7 @@ func TestGetMetricsListEmpty(t *testing.T) {
 	carbonserver := CarbonserverListener{
 		whisperData: path,
 		cacheGet:    cache.Get,
+		metrics:     &metricStruct{},
 	}
 
 	metrics, err := carbonserver.getMetricsList()
@@ -377,6 +379,7 @@ func TestGetMetricsListWithData(t *testing.T) {
 	carbonserver := CarbonserverListener{
 		whisperData: path,
 		cacheGet:    cache.Get,
+		metrics:     &metricStruct{},
 	}
 
 	fidx := fileIndex{}
@@ -418,6 +421,7 @@ func benchmarkFetchSingleMetricCommon(b *testing.B, test *FetchTest) {
 	carbonserver := CarbonserverListener{
 		whisperData: test.path,
 		cacheGet:    cache.Get,
+		metrics:     &metricStruct{},
 	}
 	// common
 

